jwst/resample/resample_spec_step.py

```diff
@@ -108,23 +111,41 @@ def _process_slit(self, input_models):
 
         Returns
         -------
-        result : `~jwst.datamodels.DrizProductModel`
+        result : `~jwst.datamodels.ImageModel`
             The resampled output, one per source
         """
-        resamp = resample_spec.ResampleSpecData(input_models, **self.drizpars)
-        drizzled_models = resamp.do_drizzle()
 
-        for model in drizzled_models:
-            model.meta.cal_step.resample = "COMPLETE"
-            model.meta.asn.pool_name = input_models.meta.pool_name
-            model.meta.asn.table_name = input_models.meta.table_name
-            update_s_region_spectral(model)
+        bb = input_models[0].meta.wcs.bounding_box
+        ((x1, x2), (y1, y2)) = bb
+        xmin = int(min(x1, x2))
+        ymin = int(min(y1, y2))
+        xmax = int(max(x1, x2))
+        ymax = int(max(y1, y2))
+        resamp = resample_spec.ResampleSpecData(input_models, **self.drizpars)
+        drizzled_models = resamp.do_drizzle(xmin=xmin, xmax=xmax,
```

Comment:
Passing xmin, xmax, etc. may have fixed the case of MIRI LRS fixed slit, but now a lot of NIRSpec MOS regtests are crashing in `do_drizzle`, which may be a result of this change.

Reply:
So I thought that only MIRI LRS data would be the type of data that was ImageModel into resample_spec_step. So only MIRI LRS data would go to the routine _process_slit.
Is that a wrong assumption ?

Reply:
That was the original design, but it looks like `_process_slit` is being called for spec3 processing of NIRSpec MOS data.

Reply:
Ah, right. I didn't notice that this was exclusive to the `process_slit` function, which only gets used if the input is a `SlitModel`. NIRSpec MOS input should be a `MultiSlitModel` and hence not go through this function. But something in the regtest run that @stscieisenhamer did using this code caused an error in `do_drizzle` when running `calwebb_spec3` on NIRSpec MOS data. I incorrectly connected the two.